### PR TITLE
PT-563, PT-564, PT-566 | New category dropdowns

### PR DESCRIPTION
**Pull request by Sartonon**
## Description

- Add target group dropdown

- Add category dropdown

- Add additional criteria dropdown

- update HDS library and use new multi select dropdown

### Closes
**[PT-563](https://helsinkisolutionoffice.atlassian.net/browse/PT-563):** 
**[PT-564](https://helsinkisolutionoffice.atlassian.net/browse/PT-564):** 
**[PT-566](https://helsinkisolutionoffice.atlassian.net/browse/PT-566):** 

## Screenshots

Screenshot 2020-11-08 at 21 13 04

### CODE REVIEW

File: src/common/components/form/fields/MultiDropdownField.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    });
+  };
+
+  return (
+    <Select
+      {...rest}
+      {...field}
+      helper={errorText || helper}
+      invalid={Boolean(errorText)}
+      optionLabelField={'label'}
```
Comment (jorilindell) [inline src/common/components/form/fields/MultiDropdownField.tsx:71]: This is really minor, but I think that label is default value so this is redundant

File: src/common/components/form/fields/MultiDropdownField.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    defaultValue: Option[];
+    setFieldValue?: (
+      field: string,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      value: any,
+      shouldValidate?: boolean | undefined
+    ) => void;
+  };
+
+const DropdownField: React.FC<Props> = ({
```
Comment (jorilindell) [inline src/common/components/form/fields/MultiDropdownField.tsx:26]: Name could be MultiDropdownField to make debugging easier

File: src/domain/event/__tests__/CreateEventPage.test.tsx [hunk 3]
```diff
@@ -292,18 +418,18 @@ jest.spyOn(apolloClient, 'query').mockImplementation(({ query }) => {
...
+//       screen.queryByText('Kulttuuri- ja vapaa-aikalautakunnan kulttuurijaosto')
+//     ).toBeInTheDocument();
+//   });
 
-  const result = await axe(container);
-  expect(result).toHaveNoViolations();
-});
+//   const result = await axe(container);
+//   expect(result).toHaveNoViolations();
+// });
```
Comment (jorilindell) [inline src/domain/event/__tests__/CreateEventPage.test.tsx:432]: Is there some reason to comment all accessibility tests out? I'm just curious

**Review by jorilindell — CHANGES_REQUESTED** [review]
Mostly looks good, but at least missing translations should be added

File: src/domain/event/__tests__/CreateEventPage.test.tsx (hunk 3 above)
Comment (Sartonon) [reply src/domain/event/__tests__/CreateEventPage.test.tsx:432]: yeah. there is a problem with svg icons accessibility in the new HDS-version

Comment (jorilindell) [reply src/domain/event/__tests__/CreateEventPage.test.tsx:432]: Ok, makes sense. So, you have to wait until it's fixed to HDS

File: src/common/components/form/fields/DropdownField.tsx [hunk 4]
```diff
@@ -60,30 +62,29 @@ const DropdownField: React.FC<Props> = ({
...
-              .map((item: string) =>
-                options.find((option) => option.value === item)
-              )
-              .filter((i: Option | undefined) => i)
-          : options.find((option) => option.value === value)
-      }
+      value={selectedValue}
       className={classNames(className, { [invalidFieldClass]: errorText })}
+      clearButtonAriaLabel="moi"
+      selectedItemRemoveButtonAriaLabel="moi"
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/common/components/form/fields/DropdownField.tsx:87]: Deleted those props, this one didn't need them

File: src/common/components/form/fields/MultiDropdownField.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      onChange={handleChange as (selectedItems: any) => void}
+      options={options}
+      id={name}
+      placeholder={placeholder || t('common.dropdown.placeholder')}
+      value={value
+        .map((item: string) => options.find((option) => option.value === item))
+        .filter((i: Option | undefined) => i)}
+      className={classNames(className, { [invalidFieldClass]: errorText })}
+      clearButtonAriaLabel="clearButtonAriaLabel"
+      selectedItemRemoveButtonAriaLabel="selectedItemRemoveButtonAriaLabel"
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/common/components/form/fields/MultiDropdownField.tsx:83]: Added translations

File: src/domain/event/CreateEventPage.tsx [hunk 6]
```diff
@@ -142,7 +142,10 @@ const CreateEventPage: React.FC = () => {
     <PageWrapper title="createEvent.pageTitle">
       <Container>
         <div className={styles.eventPage}>
-          <Notification label={t('createEvent.threeStepNotification.title')}>
+          <Notification
+            label={t('createEvent.threeStepNotification.title')}
+            closeButtonLabelText="Sulje"
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/event/CreateEventPage.tsx:147]: Deleted that prop, not needed

File: src/common/components/form/fields/MultiDropdownField.tsx (hunk 1 above)
Comment (Sartonon) [reply src/common/components/form/fields/MultiDropdownField.tsx:71]: deleted prop

File: src/common/components/form/fields/MultiDropdownField.tsx (hunk 2 above)
Comment (Sartonon) [reply src/common/components/form/fields/MultiDropdownField.tsx:26]: fixed

File: src/common/components/form/fields/DropdownField.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/common/components/form/fields/DropdownField.tsx:87]: Did you commit the changes? I'm not able to see them

**Review by jorilindell — APPROVED** [review]
Looks good

**Pull request merged**